### DISCUSSION TITLE
training Consistency Models

### DISCUSSION THREAD

**Discussion by Hedvigskorge** [huggingface/diffusers#7853] [Q&A]
Hello everyone,

Has anyone successfully utilized the CMStochasticIterativeScheduler for training a Consistency Model from scratch?

I've been exploring the pipeline_consistency_models.py, which relies on a pretrained model. However, I'm looking for examples or guidance on how to integrate this scheduler into a custom script for Consistency Training using my own dataset. particularly how to compute the loss function when integrating this scheduler.

Any insights or examples would be greatly appreciated!

**Comment by tolgacangoz** [comment]
Hello @Hedvigskorge,
Training examples can be found in the `examples` folder. [Could you examine consistency-related files?](https://github.com/search?q=repo%3Ahuggingface%2Fdiffusers+path%3Aconsistency+path%3A%2F%5Eexamples%5C%2F%2F&type=code)
